tools/precompiler/precompiler.py

```diff
@@ -24,8 +24,8 @@
 
 def _create_parser() -> "argparse.Namespace":
     parser = argparse.ArgumentParser(fromfile_prefix_chars="@")
-    parser.add_argument("--invalidation_mode")
-    parser.add_argument("--optimize", type=int)
+    parser.add_argument("--invalidation_mode", default="TIMESTAMP")
```

Comment:
We can't default to timestamp because it would cause non-deterministic behavior / cache thrashing. The timestamp of the source file will reflect when it was e.g. installed or cloned, not necessarily when it was last modified.

I think it's fine to default it to checked_hash, though. That's captures the intent of timestamp.

Reply:
Ah yes, this was the "default default" of Python itself, but for our purposes CHECKED_HASH is much better.

```suggestion
    parser.add_argument("--invalidation_mode", default="CHECKED_HASH")
```